test/integration/scheduler_perf/util.go

```diff
@@ -306,28 +306,51 @@ func (tc *throughputCollector) run(ctx context.Context) {
 		klog.Fatalf("%v", err)
 	}
 	lastScheduledCount := len(podsScheduled)
-	ticker := time.NewTicker(throughputSampleFrequency)
+	ticker := time.NewTicker(throughputSampleInterval)
 	defer ticker.Stop()
+	lastSampleTime := time.Now()
+	doSample := func() {
+		now := time.Now()
+		podsScheduled, err := getScheduledPods(tc.podInformer, tc.namespaces...)
+		if err != nil {
+			klog.Fatalf("%v", err)
+		}
+
+		scheduled := len(podsScheduled)
+		// Only do sampling if number of scheduled pods is greater than zero
+		if scheduled > 0 {
+			newScheduled := scheduled - lastScheduledCount
+			if newScheduled == 0 {
+				// Throughput would be zero for the last interval.
+				// Instead of recording 0 pods/s, keep expanding
+				// the interval until we see at least one additional
+				// pod being scheduled.
```

Comment:
This logic is flawed: it leads to samples that span different durations and calculating the percentiles is no longer accurate.

The average could be fixed by weighing each sample with its duration, but the percentiles are harder. One would have to determine a smooth function that approximates the actual samples and then sample that curve at regular intervals. Probably not worth it.

Reply:
This might not be that hard to fix after all.

Suppose one sample gets skipped because it would be zero. The next sample then covers both intervals with a total of, say, 0.5pod/s. When adding *two* samples with that value, the overall number of samples remains the same and percentiles continue to work as they did before.

Reply:
Fixed.